exchanges/poloniex/poloniex_wrapper.go

```diff
@@ -939,52 +948,42 @@ func (p *Poloniex) ValidateCredentials(ctx context.Context, assetType asset.Item
 }
 
 // GetHistoricCandles returns candles between a time period for a set time interval
-func (p *Poloniex) GetHistoricCandles(ctx context.Context, pair currency.Pair, a asset.Item, start, end time.Time, interval kline.Interval) (kline.Item, error) {
-	if err := p.ValidateKline(pair, a, interval); err != nil {
-		return kline.Item{}, err
-	}
-
-	formattedPair, err := p.FormatExchangeCurrency(pair, a)
+func (p *Poloniex) GetHistoricCandles(ctx context.Context, pair currency.Pair, a asset.Item, interval kline.Interval, start, end time.Time) (*kline.Item, error) {
+	req, err := p.GetKlineRequest(pair, a, interval, start, end)
 	if err != nil {
-		return kline.Item{}, err
+		return nil, err
 	}
 
-	// we use Truncate here to round star to the nearest even number that matches the requested interval
-	// example 10:17 with an interval of 15 minutes will go down 10:15
-	// this is due to poloniex returning a non-complete candle if the time does not match
-	candles, err := p.GetChartData(ctx,
-		formattedPair.String(),
-		start.Truncate(interval.Duration()), end,
-		p.FormatExchangeKlineInterval(interval))
+	// we use Truncate here to round start to the nearest even number that
+	// matches the requested interval example 10:17 with an interval of 15
+	// minutes will go down 10:15 this is due to poloniex returning a
+	// non-complete candle if the time does not match.
+	resp, err := p.GetChartData(ctx,
+		req.RequestFormatted.String(),
+		req.Start.Truncate(req.ExchangeInterval.Duration()),
```

Comment:
```suggestion
	resp, err := p.GetChartData(ctx,
		req.RequestFormatted.String(),
		req.Start,
```
You now truncate in getklinerequest